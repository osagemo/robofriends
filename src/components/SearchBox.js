import React from "react";

export default function SearchBox({ searchField, searchChange }) {
  console.log("SearchBox");
  return (
    <div className="pa2">
      <input
        aria-label="Search robots"
        className="pa3 ba b--green bg-lightest-blue"
        type="search"
        placeholder="search robots"
        onChange={searchChange}
      />
    </div>
  );
}
