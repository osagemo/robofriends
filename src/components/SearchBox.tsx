import React from "react";
import { RoboActionTypes } from "../types";

interface ISearchBoxProps {
  onSearchChange: (
    event: React.ChangeEvent<HTMLInputElement>
  ) => RoboActionTypes;
}

const SearchBox: React.FC<ISearchBoxProps> = ({ onSearchChange }) => {
  return (
    <div className="pa2">
      <input
        aria-label="Search robots"
        className="pa3 ba b--green bg-lightest-blue"
        type="search"
        placeholder="search robots"
        onChange={onSearchChange}
      />
    </div>
  );
};

export default SearchBox;
