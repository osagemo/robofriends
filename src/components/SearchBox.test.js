import React from "react";
import { shallow } from "enzyme";
import SearchBox from "./SearchBox";

// setSearchField(event.target.value)

let wrapper;
let searchField;
const searchChange = text => {
  searchField = text;
};

beforeEach(() => {
  wrapper = shallow(<SearchBox searchChange={searchChange} />);
});

it("renders correctly", () => {
  expect(wrapper).toMatchSnapshot();
});

it("sets a searchField from text input onChange", () => {
  wrapper.find("input").simulate("change", "hello");
  expect(searchField).toEqual("hello");
});
