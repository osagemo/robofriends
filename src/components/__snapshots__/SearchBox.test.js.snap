// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="pa2"
>
  <input
    aria-label="Search robots"
    className="pa3 ba b--green bg-lightest-blue"
    onChange={[Function]}
    placeholder="search robots"
    type="search"
  />
</div>
`;
